### ISSUE TITLE
cflags from pkg-config are not fully de-duplicated

### ISSUE THREAD

**Issue by DyXel** [mesonbuild/meson#13803]
**Describe the bug**
Duplicated compiler arguments are printed whenever a compiler error is produced, the duplicated arguments come from the output of `pkg-config --cflags {dependency}`. _Some_ arguments are de-duplicated properly, for example, `-DNOMINMAX`.

**To Reproduce**
Example with `protobuf` as dependency, it pulls the flags from all the `absl` bits listed in the `Requires` section, resulting in a ton of `-Wnon-virtual-dtor` and `-DNOMINMAX` (the latter which are filtered properly).
`meson.build`:
```
project('pctest', 'cpp')
executable('pctest', 'main.cpp', dependencies : dependency('protobuf'))
```
`main.cpp`:
```
#error "test"
```

![-Wnon-virtual-dtor-cascade](https://github.com/user-attachments/assets/d26f3463-ea1d-4447-9b4c-760fc4baffb9)

**Expected behavior**
I would expect the `-Wnon-virtual-dtor` to be outputted only once per error on object / target.

**System parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)
  * Plain build.
* What operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
  * Arch Linux
* What Python version are you using e.g. 3.8.0
  * Python 3.12.7
* What `meson --version`
  * 1.5.2
* What `ninja --version` if it's a Ninja build
  * 1.12.1

**Comment by bonzini** [comment]
Indeed. this is not great but it's because -Wfoo -Wno-foo -Wfoo should *not* unify the flags to -Wfoo -Wno-foo.

**Comment by DyXel** [comment]
I am not sure I understand why this can't or won't be fixed? Is there anything preventing from collapsing compiler warning options. At least GCC has [documentation](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#Options-to-Request-or-Suppress-Warnings) on how they handle them. e.g.:

 * `-Wnon-virtual-dtor  -Wno-non-virtual-dtor -Wnon-virtual-dtor` -> `-Wnon-virtual-dtor`

Bonus:

 * `-Wnon-virtual-dtor -Wnon-virtual-dtor -Wnon-virtual-dtor -Wnon-virtual-dtor` -> `-Wnon-virtual-dtor`
 * `-Wnon-virtual-dtor -Wnon-virtual-dtor -Wno-non-virtual-dtor` -> `-Wno-non-virtual-dtor`
